test/tests/mysql-basics/run.sh

```diff
@@ -22,10 +22,16 @@ cid="$(
 )"
 trap "docker rm -vf $cid > /dev/null" EXIT
 
+if docker run --rm "$image" mariadb --version > /dev/null; then
+	mysql_cli=mariadb
+else
+	mysql_cli=mysql
+fi
+
 mysql() {
 	docker run --rm -i \
 		--link "$cname":mysql \
-		--entrypoint mysql \
+		--entrypoint "$mysql_cli" \
 		-e MYSQL_PWD="$MYSQL_PASSWORD" \
 		"$image" \
 		-hmysql \
```

Comment:
If we munge this a little bit, we can avoid the variable and the extra `docker run` entirely: :eyes:

```suggestion
		--entrypoint sh \
		-e MYSQL_PWD="$MYSQL_PASSWORD" \
		"$image" \
		-euc 'if command -v mariadb > /dev/null; then exec mariadb "$@"; else exec mysql "$@"; fi' -- \
		-hmysql \
```